project-koku/masu
# Updating summary tables only for specific provider

Addresses #383 

Now OCP summary is done on a `cluster_id` basis and it is assuming that there is always a 1-1 mapping between OCP provider and OCP cluster.

The reason for this change is to ensure that OCP charge calculations are applied only to the appropriate cluster.

**Testing**
1.  Add OCP provider with rates
2. Ingest Provider with rates and verify koku API that derived_cost is populated
3.  Add a second OCP provider without rates and the same report data as the other provider.
4. Ingest and veirfy koku API that my-ocp-cluster-2 has derived_cost=0 and my-ocp-cluster-1 is unchanged
5. Add cpu/mem rates to provider 2 and run the update_charge endpoint on masu
6. Sanity check openshift/costs endpoint and make sure that my-ocp-cluster-2 does not have any storage costs.
7. Sanity check OCP capacity calculation.  capacity field in meta-total section should add up to what the two clusters have for capacity.

**Test Results**

[masu_383_ut.txt](https://github.com/project-koku/masu/files/2971248/masu_383_ut.txt)
